File: API/Entities/Course.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{
    public class Course
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public int Duration { get; set; } = 1;
        public int PriceFull { get; set; }
        public int PriceMonthly { get; set; }
        public string ImgUrl { get; set; }
        public string PublicId { get; set; }
        public List<Section> Sections { get; set; }
    }
}

File: client/src/app/models/course.ts
import { Lesson, LessonPreview } from "./lesson"

export interface Course {
  id: number
  title: string
  description: string
  duration: number
  priceFull: number
  priceMonthly: number
  sections: Section[]
}

export interface Section {
  id: number
  number: number
  title: string
  description: string
  isAvailable: boolean
  lessons: Lesson[]
}

export interface CoursePreview {
  id: number
  title: string
  description: string
  duration: number
  priceFull: number
  priceMonthly: number
  sections: SectionPreview[]
}

export interface SectionPreview {
  id: number
  number: number
  title: string
  description: string
  isAvailable: number
  lessons: LessonPreview[]
}